pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,661 @@
+	clusterWorkspaceInformer := kcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces().Informer()
+	crInformer := kubeSharedInformerFactory.Rbac().V1().ClusterRoles().Informer()
+	crbInformer := kubeSharedInformerFactory.Rbac().V1().ClusterRoleBindings().Informer()
+
+	_ = addWorkspaceFullPathIndexer(clusterWorkspaceInformer)
+
+	return &homeWorkspaceHandler{
+		apiHandler:    apiHandler,
+		authz:         a,
+		creationDelay: creationDelay,
+		homePrefix:    homePrefix,
+		bucketLevels:  bucketLevels,
+		bucketSize:    bucketSize,
+		getLocalClusterWorkspace: func(fullName logicalcluster.Name) (*tenancyv1alpha1.ClusterWorkspace, error) {
+			result, err := clusterWorkspaceInformer.GetIndexer().ByIndex(workspaceFullPathIndexName, fullName.String())
+			if err != nil {
+				return nil, err
+			}
+			if len(result) == 0 {
+				return nil, nil
+			}
+			return result[0].(*tenancyv1alpha1.ClusterWorkspace), nil
+		},
+		searchLocalClusterRole: func(workspace logicalcluster.Name, name string) (bool, error) {
+			_, ok, err := crInformer.GetIndexer().GetByKey(clusters.ToClusterAwareKey(workspace, name))
+			return ok, err
+		},
+		searchLocalClusterRoleBinding: func(workspace logicalcluster.Name, name string) (bool, error) {
+			_, ok, err := crbInformer.GetIndexer().GetByKey(clusters.ToClusterAwareKey(workspace, name))
+			return ok, err
+		},
+		synced: func() bool {
+			return clusterWorkspaceInformer.HasSynced() &&
+				crInformer.HasSynced() &&
+				crbInformer.HasSynced()
+		},
+		createClusterRole: func(ctx context.Context, workspace logicalcluster.Name, cr *rbacv1.ClusterRole) error {
+			_, err := kubeClusterClient.Cluster(workspace).RbacV1().ClusterRoles().Create(ctx, cr, metav1.CreateOptions{})
+			return err
+		},
+		createClusterRoleBinding: func(ctx context.Context, workspace logicalcluster.Name, crb *rbacv1.ClusterRoleBinding) error {
+			_, err := kubeClusterClient.Cluster(workspace).RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
+			return err
+		},
+		createClusterWorkspace: func(ctx context.Context, workspace logicalcluster.Name, cw *tenancyv1alpha1.ClusterWorkspace) error {
+			_, err := kcpClusterClient.Cluster(workspace).TenancyV1alpha1().ClusterWorkspaces().Create(ctx, cw, metav1.CreateOptions{})
+			return err
+		},
+		getClusterWorkspace: func(ctx context.Context, workspace logicalcluster.Name, name string) (*tenancyv1alpha1.ClusterWorkspace, error) {
+			return kcpClusterClient.Cluster(workspace).TenancyV1alpha1().ClusterWorkspaces().Get(ctx, name, metav1.GetOptions{})
+		},
+	}
+}
+
+func addWorkspaceFullPathIndexer(clusterWorkspaceInformer cache.SharedIndexInformer) error {
```

Comment:
move to indexes.go

Reply:
why is this index needed in the first place? it is doing the same as looking up by `clusters.ToClusterAwareKey(clusterName, clusterWorkspaceName)`

Reply:
I removed the indexer which was useless, as you pointed.
Fixed in commit  eac1f8863bdcae327ee37064046c35c46e2bc562